packages/apps/reputation-oracle/server/src/modules/user/user.error.filter.ts

```diff
@@ -0,0 +1,44 @@
+import {
+  ExceptionFilter,
+  Catch,
+  ArgumentsHost,
+  HttpStatus,
+  Logger,
+} from '@nestjs/common';
+import { Request, Response } from 'express';
+
+import {
+  UserError,
+  DuplicatedWalletAddressError,
+  InvalidWeb3SignatureError,
+} from './user.error';
+
+type UserControllerError =
+  | UserError
+  | DuplicatedWalletAddressError
+  | InvalidWeb3SignatureError;
+
+@Catch(UserError, DuplicatedWalletAddressError, InvalidWeb3SignatureError)
+export class UserErrorFilter implements ExceptionFilter {
+  private logger = new Logger(UserErrorFilter.name);
+  catch(exception: UserControllerError, host: ArgumentsHost) {
+    const ctx = host.switchToHttp();
+    const response = ctx.getResponse<Response>();
+    const request = ctx.getRequest<Request>();
+    let status = HttpStatus.BAD_REQUEST;
+
+    if (exception instanceof DuplicatedWalletAddressError) {
+      status = HttpStatus.CONFLICT;
+    } else if (exception instanceof InvalidWeb3SignatureError) {
+      status = HttpStatus.CONFLICT;
```

Comment:
Maybe we change it to `400`? I checked the UI and don't see us relying on 409 for signature flow